test/metabase/models/collection_test.clj

```diff
@@ -1556,4 +1568,34 @@
                              :name     "a"
                              :location "/3/"
                              :children []}]}]
-               (collection/collections->tree collections)))))))
+               (collection/collections->tree {} collections)))))))
+
+(deftest annotate-collections-test
+  (let [collections [{:id 1, :name "a", :location "/"}
+                     {:id 2, :name "b", :location "/1/"}
+                     {:id 3, :name "c", :location "/1/2/"}
+                     {:id 4, :name "d", :location "/1/2/3/"}
+                     {:id 5, :name "e", :location "/1/"}]
+        clean      #(walk/prewalk
+                     (fn [x]
+                       ;; select important keys and remove empty children
+                       (if (map? x)
+                         (cond-> (select-keys x [:id :here :below :children])
+                           (not (seq (:children x))) (dissoc :children))
+                         x))
+                     %)]
+    (is (= [{:id 1 :name "a" :location "/" :here #{:card} :below #{:card :dataset}}
+            {:id 2, :name "b", :location "/1/" :below #{:dataset}}
+            {:id 3, :name "c", :location "/1/2/" :here #{:dataset} :below #{:dataset}}
+            {:id 4, :name "d", :location "/1/2/3/" :here #{:dataset}}
+            {:id 5, :name "e", :location "/1/" :here #{:card}}]
+           (collection/annotate-collections {:card #{1 5} :dataset #{3 4}} collections)))
+    (is (= [{:id 1 :here #{:card} :below #{:card :dataset}
```

Comment:
nit: slightly hard to follow the structure of this map literal.  Does it make sense to align map values and keep every kv pair on a separate line (i.e. do what's done with `:children` below for every entry)?

Reply:
Good point. I'll play with the alignment a little. I bias towards code width since i'm split screen buffers all the time. I'll clean it up so its super readable